tools/serverpod_cli/bin/generator/class_analyzer.dart

```diff
@@ -523,3 +608,22 @@ class ClassAnalyzer {
     return true;
   }
 }
+
+extension on TypeDefinition {
```

Comment:
> Any reason this is an extension, rather than part of TypeDefinition itself?

Originally it was a top level function. Then I changed it to be an extension. So it's a historic thing. I changed it in 11e0ac3

> It seems like this should perhaps be part of the constructor?

At that time we don't have the `classDefinitions` generated, yet.